### PR TITLE
[BEAM-2666] Add JStorm runner to Beam asf-site

### PR DESCRIPTION
**Pull request by bastiliu**

### CODE REVIEW

**Comment by bastiliu** [comment]
@peihe

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/680/

Jenkins built the site at commit id 0e033a64000a4c33dda912ae35537d28b7ab34e0 with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/315/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

File: src/documentation/runners/jstorm.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+permalink: /documentation/runners/jstorm/
+redirect_from: /learn/runners/jstorm/
+---
+# Using the JStorm Runner
+
+The JStorm Runner can be used to execute Beam pipelines using [JStorm](http://jstorm.io/). 
+
+The JStorm Runner executes Beam pipelines on top of JStorm, providing:
+
+* The at-least-once and exactly-once fault tolerance.
```
Comment (peihe) [inline src/documentation/runners/jstorm.md]: add "high throughput and low latency"
and more about standalone and integration with Hadoop

File: src/documentation/runners/jstorm.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+### Deploying JStorm with your application
+
+To run against a Standalone cluster:
+you can package your program with all beam dependencies into a fat jar, and then submit the topology by following command.
+```
+jstorm jar WordCount.jar org.apache.beam.examples.WordCount --Runner=org.apache.beam.runners.jstorm.JStormRunner
+```
+
+For the case that if you don't want to package a fat jar, you can upload the beam dependencies onto all cluster nodes($JSTORM_HOME/lib/ext/beam) first.
+When you submit topology with argument **"--external-libs beam"**, JStorm will load the beam dependencies automatically.
```
Comment (peihe) [inline src/documentation/runners/jstorm.md]: \`--external-libs beam\`

File: src/documentation/runners/jstorm.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+### Deploying JStorm with your application
+
+To run against a Standalone cluster:
+you can package your program with all beam dependencies into a fat jar, and then submit the topology by following command.
+```
+jstorm jar WordCount.jar org.apache.beam.examples.WordCount --Runner=org.apache.beam.runners.jstorm.JStormRunner
+```
+
+For the case that if you don't want to package a fat jar, you can upload the beam dependencies onto all cluster nodes($JSTORM_HOME/lib/ext/beam) first.
```
Comment (peihe) [inline src/documentation/runners/jstorm.md]: \`$JSTORM_HOME/lib/ext/beam\`

File: src/documentation/runners/jstorm.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  <version>{{ site.release_latest }}</version>
+</dependency>
+```
+
+### Deploying JStorm with your application
+
+To run against a Standalone cluster:
+you can package your program with all beam dependencies into a fat jar, and then submit the topology by following command.
+```
+jstorm jar WordCount.jar org.apache.beam.examples.WordCount --Runner=org.apache.beam.runners.jstorm.JStormRunner
```
Comment (peihe) [inline src/documentation/runners/jstorm.md]: --runner?

File: src/documentation/runners/jstorm.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+For the case that if you don't want to package a fat jar, you can upload the beam dependencies onto all cluster nodes($JSTORM_HOME/lib/ext/beam) first.
+When you submit topology with argument **"--external-libs beam"**, JStorm will load the beam dependencies automatically.
+```
+jstorm jar WordCount.jar org.apache.beam.examples.WordCount --external-libs beam  --Runner=org.apache.beam.runners.jstorm.JStormRunner
```
Comment (peihe) [inline src/documentation/runners/jstorm.md]: --runner?

File: src/documentation/runners/jstorm.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  <td>false</td>
+</tr>
+<tr>
+  <td><code>localMode</code></td>
+  <td>Indicate if the topology is running on local machine or distributed cluster</td>
+  <td>false</td>
+</tr>
+<tr>
+  <td><code>localModeExecuteTimeSec</code></td>
+  <td>Executing time(sec) of topology on local mode. Default is 1 min</td>
```
Comment (peihe) [inline src/documentation/runners/jstorm.md]: remove " Default is 1 min"

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/688/

Jenkins built the site at commit id cd8bbcb93dc0fae7d2a279f8832fcf5dedbaab2d with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/315/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

**Comment by peihe** [comment]
LGTM

+R: @kennknowles

**Comment by kennknowles** [comment]
R: @melap 

On the overview page, we just list the runners that are on master. In the capability matrix, runners on feature branches should be marked as "in development" with details saying that they aren't part of the release.

What is the license of that JStorm logo?

**Comment by bastiliu** [comment]
@kennknowles  No special license for the logo. The whole project of JStorm is under Apache license.

File: src/documentation/runners/jstorm.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
+---
+layout: default
+title: "JStorm Runner"
+permalink: /documentation/runners/jstorm/
+redirect_from: /learn/runners/jstorm/
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: the redirect_from line shouldn't be needed as this page is new

File: src/documentation/runners/jstorm.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+Like a native JStorm topology, user can execute Beam topology on local mode, standalone cluster or jstorm-on-yarn cluster.
+
+The [Beam Capability Matrix]({{ site.baseurl }}/documentation/runners/capability-matrix/) documents the currently supported capabilities of the JStorm Runner.
+
+## JStorm Runner prerequisites and setup
+
+The JStorm runner currently supports JStorm 2.5.0-SNAPSHOT version.
+
+You can add a dependency on the latest version of the JStorm runner by adding to your pom.xml the following:
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: ... by adding the following to your pom.xml.

File: src/documentation/runners/jstorm.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+* High throughput and low latency.
+* The at-least-once and exactly-once fault tolerance.
+
+Like a native JStorm topology, user can execute Beam topology on local mode, standalone cluster or jstorm-on-yarn cluster.
+
+The [Beam Capability Matrix]({{ site.baseurl }}/documentation/runners/capability-matrix/) documents the currently supported capabilities of the JStorm Runner.
+
+## JStorm Runner prerequisites and setup
+
+The JStorm runner currently supports JStorm 2.5.0-SNAPSHOT version.
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: JStorm version 2.5.0-SNAPSHOT.

File: src/documentation/runners/jstorm.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+<dependency>
+  <groupId>org.apache.beam</groupId>
+  <artifactId>beam-runners-jstorm</artifactId>
+  <version>{{ site.release_latest }}</version>
+</dependency>
+```
+
+### Deploying JStorm with your application
+
+To run against a Standalone cluster, you can package your program with all beam dependencies into a fat jar, and then submit the topology by following command.
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: capitalize Beam
"by the following" -> "with the following"

File: src/documentation/runners/jstorm.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+redirect_from: /learn/runners/jstorm/
+---
+# Using the JStorm Runner
+
+The JStorm Runner can be used to execute Beam pipelines using [JStorm](http://jstorm.io/), while providing:
+
+* High throughput and low latency.
+* The at-least-once and exactly-once fault tolerance.
+
+Like a native JStorm topology, user can execute Beam topology on local mode, standalone cluster or jstorm-on-yarn cluster.
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: ... users can execute a Beam topology with local mode ...

File: src/documentation/runners/jstorm.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+title: "JStorm Runner"
+permalink: /documentation/runners/jstorm/
+redirect_from: /learn/runners/jstorm/
+---
+# Using the JStorm Runner
+
+The JStorm Runner can be used to execute Beam pipelines using [JStorm](http://jstorm.io/), while providing:
+
+* High throughput and low latency.
+* The at-least-once and exactly-once fault tolerance.
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: suggest removing "The" --> "At-least-once and exactly-once fault tolerance."

File: src/documentation/runners/jstorm.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+### Deploying JStorm with your application
+
+To run against a Standalone cluster, you can package your program with all beam dependencies into a fat jar, and then submit the topology by following command.
+```
+jstorm jar WordCount.jar org.apache.beam.examples.WordCount --runner=org.apache.beam.runners.jstorm.JStormRunner
+```
+
+For the case that if you don't want to package a fat jar, you can upload the beam dependencies onto all cluster nodes(`$JSTORM_HOME/lib/ext/beam`) first.
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: suggest removing "For the case that"  --> "If you don't want to package ..."
capitalize Beam

File: src/documentation/runners/jstorm.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+When you submit topology with argument `"--external-libs beam"`, JStorm will load the beam dependencies automatically.
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: submit a topology
capitalize Beam

File: src/documentation/runners/jstorm.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+```
+jstorm jar WordCount.jar org.apache.beam.examples.WordCount --external-libs beam  --runner=org.apache.beam.runners.jstorm.JStormRunner
+```
+
+For the details of deploying JStorm cluster, please refer to [JStorm cluster deploy](http://jstorm.io/QuickStart/Deploy/index.html)
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: what about something like: "To learn about deploying a JStorm cluster, ..."

File: src/documentation/runners/jstorm.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+  <td>Executing time(sec) of topology on local mode.</td>
+  <td>60</td>
+</tr>
+</table>
+
+## Additional notes
+
+### Monitoring your job
+You can monitor your job on JStorm UI while all JStorm system metrics and beam metrics can be displayed on JStorm UI.
+For testing on local mode, you can retreive the beam metrics by the metric API of PipelineResult.
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: capitalize Beam
"... with the `metrics` method of `PipelineResult`."

File: src/documentation/runners/jstorm.md [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+  <td><code>localModeExecuteTimeSec</code></td>
+  <td>Executing time(sec) of topology on local mode.</td>
+  <td>60</td>
+</tr>
+</table>
+
+## Additional notes
+
+### Monitoring your job
+You can monitor your job on JStorm UI while all JStorm system metrics and beam metrics can be displayed on JStorm UI.
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: Is this still accurate?
You can monitor your job with the JStorm UI, which displays all JStorm system metrics and Beam metrics.

File: src/documentation/runners/jstorm.md (hunk 7 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/documentation/runners/jstorm.md (hunk 1 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/documentation/runners/jstorm.md (hunk 9 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/documentation/runners/jstorm.md (hunk 8 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/documentation/runners/jstorm.md (hunk 4 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/documentation/runners/jstorm.md (hunk 2 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/documentation/runners/jstorm.md (hunk 5 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/documentation/runners/jstorm.md (hunk 15 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/documentation/runners/jstorm.md (hunk 6 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/documentation/runners/jstorm.md (hunk 17 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done.

File: src/documentation/runners/jstorm.md (hunk 16 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

**Comment by bastiliu** [comment]
@melap  Thanks for your review. I have addressed all comments.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/700/

Jenkins built the site at commit id 8e71e425e6bcbed3d9f227f4f7988511169722c3 with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/315/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

File: src/documentation/runners/jstorm.md [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+title: "JStorm Runner"
+---
+# Using the JStorm Runner
+
+The JStorm Runner can be used to execute Beam pipelines using [JStorm](http://jstorm.io/), while providing:
+
+* High throughput and low latency.
+* At-least-once and exactly-once fault tolerance.
+
+Like a native JStorm topology, user can execute Beam topology with local mode, standalone cluster or jstorm-on-yarn cluster.
```
Comment (melap) [inline src/documentation/runners/jstorm.md]: user -> users

File: src/get-started/beam-overview.md [hunk 19]
```diff
@@ -40,6 +40,8 @@ Beam currently supports Runners that work with the following distributed process
          alt="Apache Spark">
 * Google Cloud Dataflow <img src="{{ site.baseurl }}/images/logos/runners/dataflow.png"
          alt="Google Cloud Dataflow">
+* JStorm <img src="{{ site.baseurl }}/images/logos/runners/jstorm.png"
```
Comment (melap) [inline src/get-started/beam-overview.md]: I think as per @kennknowles only runners in master should be on the overview page, so this should be removed.

File: src/documentation/runners/jstorm.md (hunk 18 above)
Comment (bastiliu) [reply src/documentation/runners/jstorm.md]: Done

File: src/get-started/beam-overview.md (hunk 19 above)
Comment (bastiliu) [reply src/get-started/beam-overview.md]: Done

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/707/

Jenkins built the site at commit id 4053b4cf1681c96fb1ed193a586c700ab8f15db5 with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/315/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

**Comment by kennknowles** [comment]
This looks good to me. It has conflicts with the MapReduce runner bits that I just merged. And both runners need links from the "Ongoing Projects" page to these detail pages that you added.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/716/

Jenkins built the site at commit id 08618ae39ee0a9197eb5a848d459a3ece05dccd5 with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/315/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

**Comment by bastiliu** [comment]
@kennknowles Thanks. The link in "Ongoing Projects" was added.

File: src/contribute/work-in-progress.md [hunk 20]
```diff
@@ -25,7 +25,7 @@ Current branches include:
 | Feature | Branch | JIRA Component | More Info |
 | ---- | ---- | ---- | ---- |
 | Apache Spark 2.0 Runner | [runners-spark2](https://github.com/apache/beam/tree/runners-spark2) | - | [thread](https://lists.apache.org/thread.html/e38ac4e4914a6cb1b865b1f32a6ca06c2be28ea4aa0f6b18393de66f@%3Cdev.beam.apache.org%3E) |
-| JStorm Runner | [jstorm-runner](https://github.com/apache/beam/tree/jstorm-runner) | [runner-jstorm](https://issues.apache.org/jira/browse/BEAM/component/12332477) | [BEAM-1899](https://issues.apache.org/jira/browse/BEAM-1899) |
+| [JStorm Runner]({{ site.baseurl }}/documentation/runners/jstorm.md) | [jstorm-runner](https://github.com/apache/beam/tree/jstorm-runner) | [runner-jstorm](https://issues.apache.org/jira/browse/BEAM/component/12332477) | [BEAM-1899](https://issues.apache.org/jira/browse/BEAM-1899) |
```
Comment (kennknowles) [inline src/contribute/work-in-progress.md]: Drop the `.md` I think, for this link to work. (check the Jenkins preview version, and the other files in the repo)

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/722/

Jenkins built the site at commit id d625c64f0b6fe7fe143866d78bc2c55ec6e4ad5f with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/315/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/723/

Jenkins built the site at commit id be9abb0147c97c2e6b152580047197666ffbd7d5 with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/315/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

File: src/contribute/work-in-progress.md (hunk 20 above)
Comment (bastiliu) [reply src/contribute/work-in-progress.md]: Done. It works now.

**Comment by kennknowles** [comment]
Great! LGTM. Can you squash the commits? Then I will merge, or Pei can.

**Comment by bastiliu** [comment]
Thanks keen. Pei will help to merge.

**Comment by bastiliu** [comment]
@peihe  Squash was done.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/729/

Jenkins built the site at commit id 2cf192c9cd2c8642a3367f0150d860740ede5258 with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/315/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

**Comment by peihe** [comment]
@asfgit merge

**Pull request merged**